docs/architecture/transactions/execution.md

```diff
@@ -1,38 +1,48 @@
-# Transaction Execution
-Transactions are being executed by the Miden Transaction Executor. Transaction execution results in a `ExecutedTransaction` object and consists of the following steps:
+The Miden transaction executor is the component that executes transactions. 
```

Comment:
I don't know how yet, but we could also try embedding this explanation here (https://crates.io/crates/miden-tx) into this chapter. 

My explanation has more information, but I like how concise the explanation on crate.io is. Also, when we show this code, it might be easier for some readers to follow.

What do you think?

------8<------


The first requirement is to have a DataStore implementation. DataStore objects are responsible to load the data needed by the transactions executor, specially the account's code, the reference block data, and the note's inputs.

```
let store = DataStore:new();
```

Once a store is available, a TransactionExecutor object can be used to execute a transaction. Consuming a zero or more notes, and possibly calling some of the account's code.

```
let executor = TransactionExecutor::new(store);
let executed_transaction = executor.execute_transaction(account_id, block_ref, note_ids, tx_args);
```

With the transaction execution done, it is then possible to create a proof:

```
let prover = TransactionProver::new(ProvingOptions::default());
let proven_transaction = prover.prove_transaction(executed_transaction);
```

And to verify a proof:

```
let verifier = TransactionVerifier::new(SECURITY_LEVEL);
verifier.verify(proven_transaction);
```

Reply:
this is more of an API doc so I would add it separately in an API section of the docs, not in the architecture section ... let me know and i can do that